src/runtime/classobj.cpp

```diff
@@ -735,6 +735,182 @@ static Box* instanceNext(BoxedInstance* inst) {
     return r;
 }
 
+static PyObject* generic_binary_op(PyObject* v, PyObject* w, char* opname) {
+    PyObject* result;
+    PyObject* args;
+    PyObject* func = PyObject_GetAttrString(v, opname);
+    if (func == NULL) {
+        if (!PyErr_ExceptionMatches(PyExc_AttributeError))
+            return NULL;
+        PyErr_Clear();
+        Py_INCREF(Py_NotImplemented);
+        return Py_NotImplemented;
+    }
+    args = PyTuple_Pack(1, w);
+    if (args == NULL) {
+        Py_DECREF(func);
+        return NULL;
+    }
+    result = PyEval_CallObject(func, args);
+    Py_DECREF(args);
+    Py_DECREF(func);
+    return result;
+}
+
```

Comment:
Is that ok to copy these code? Or use a "Pyston way" function? Such as `generic_binary_op` could replaced by `_instanceBinary` function. But for other functions, such as `half_binop` and `do_pow`, it contains logic. So either rewrite it ( like the last version of commit), or call the CPython version.